bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java

```diff
@@ -0,0 +1,349 @@
+        stopNormalPoll();
+        stopRapidPoll();
+    }
+
+    private synchronized void stopNormalPoll() {
+        stopFuture(normalPollFuture);
+        normalPollFuture = null;
+    }
+
+    private synchronized void stopRapidPoll() {
+        stopFuture(rapidPollFuture);
+        rapidPollFuture = null;
+    }
+
+    private void stopFuture(@Nullable Future<?> future) {
+        if (future != null) {
+            future.cancel(true);
+        }
+    }
+
+    private synchronized void restartPolls(boolean rapid) {
+        stopPolls();
+        if (rapid) {
+            normalPollFuture = scheduler.scheduleWithFixedDelay(this::normalPoll, 35, normalRefreshSeconds,
+                    TimeUnit.SECONDS);
+            rapidPollFuture = scheduler.scheduleWithFixedDelay(this::rapidPoll, 3, RAPID_REFRESH_SECONDS,
+                    TimeUnit.SECONDS);
+        } else {
+            normalPollFuture = scheduler.scheduleWithFixedDelay(this::normalPoll, 0, normalRefreshSeconds,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    private void normalPoll() {
+        stopRapidPoll();
+        fetchData();
+    }
+
+    private void rapidPoll() {
+        fetchData();
+    }
+
+    private synchronized void fetchData() {
+        try {
+            if (securityToken == null) {
+                login();
+                if (securityToken != null) {
+                    getAccount();
+                }
+            }
+
+            if (securityToken != null) {
+                getDevices();
+            }
+        } catch (InterruptedException e) {
+        }
+    }
+
+    private void login() throws InterruptedException {
+        HttpResult result = sendRequest(BASE_URL + "/v5/Login", HttpMethod.POST, null,
+                new StringContentProvider(gsonUpperCase.toJson(new LoginRequestDTO(username, password))),
+                "application/json");
+        if (HttpStatus.isSuccess(result.responseCode)) {
+            LoginResponseDTO loginResponse = gsonUpperCase.fromJson(result.content, LoginResponseDTO.class);
+            if (loginResponse != null) {
+                securityToken = loginResponse.securityToken;
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Invalid Response Body");
+            }
+        } else {
+            handleErrorResponse(result);
+            if (thing.getStatusInfo().getStatusDetail() == ThingStatusDetail.CONFIGURATION_ERROR) {
+                // bad credentials, stop trying to login
+                stopPolls();
+            }
+        }
+    }
+
+    private void getAccount() throws InterruptedException {
+        HttpResult result = sendRequest(BASE_URL + "/v5/My?expand=account", HttpMethod.GET, securityToken, null, null);
+        if (HttpStatus.isSuccess(result.responseCode)) {
+            account = gsonUpperCase.fromJson(result.content, AccountDTO.class);
```

Comment:
You should catch the JsonSyntaxException that is thrown here.

Reply:
Done.